common/djangoapps/third_party_auth/tests/specs/base.py

```diff
@@ -130,11 +130,10 @@ def assert_json_failure_response_is_inactive_account(self, response):
 
     def assert_json_failure_response_is_missing_social_auth(self, response):
         """Asserts failure on /login for missing social auth looks right."""
-        self.assertContains(
-            response,
-            u"successfully signed in to your %s account, but this account isn&#39;t linked" % self.provider.name,
-            status_code=403,
-        )
+        self.assertEqual(403, response.status_code)
+        payload = json.loads(response.content.decode('utf-8'))
+        self.assertFalse(payload.get('success'))
+        self.assertEqual(payload.get('error_code'), 'third-party-auth-with-no-linked-account')
```

Comment:
**Reviewer Note**: This is a test improvement that could have been done as part of an earlier PR.